src/intrinsics/mod.rs

```diff
@@ -41,11 +42,23 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             }));
         }
 
+        // Overwrite some core engine intrinsics that are used to catch UB -- we have better
```

Comment:
Shouldn't we remove them from the core engine and add them to the ctfe intrinsic handler if miri doesn't want them?

Reply:
I like this variant as it keeps the checks in miri const eval as you said

Reply:
Oh, because we don't do validation in normal const eval? So if we removed these checks for all interpreters, we'd miss even more UB in const eval?

Reply:
> Oh, because we don't do validation in normal const eval? So if we removed these checks for all interpreters, we'd miss even more UB in const eval?

Exactly.